dipy/direction/bingham.py

```diff
@@ -0,0 +1,772 @@
+    using the method described by Riffert et al [1]_.
+
+    References
+    ----------
+    .. [1] Riffert TW, Schreiber J, Anwander A, Knösche TR. Beyond fractional
+            anisotropy: Extraction of bundle-specific structural metrics from
+            crossing fiber models. NeuroImage. 2014 Oct 15;100:176-91.
+    """
+    # extract all maxima on the ODF
+    directions, values, _ = peak_directions(odf, sphere,
+                                            relative_peak_threshold=rel_th,
+                                            min_separation_angle=min_sep_angle)
+
+    # n becomes the new limit of peaks and sets a maximum of peaks in case
+    # the voxel has more than npeaks.
+    n = min(npeaks, values.shape[0])
+
+    # Calculate dispersion on all and each of the peaks up to 'n'
+    if values.shape[0] != 0:
+        fits = []
+        for i in range(n):
+            fit = _bingham_fit_peak(odf, directions[i], sphere,
+                                    max_search_angle)
+            fits.append(fit)
+
+    return fits, n
+
+
+def bingham_odf(f0, k1, k2, major_axis, minor_axis, vertices):
+    r"""
+    Sample a Bingham function on the directions described by `vertices`.
+
+    Parameters
+    ----------
+    f0: float
+        Maximum value of the Bingham function.
+    k1: float
+        Concentration along major axis.
+    k2: float
+        Concentration along minor axis.
+    major_axis: ndarray (3)
+        Direction of major axis
+    minor_axis: ndarray (3)
+        Direction of minor axis
+    vertices: ndarray (N, 3)
+        Unit sphere directions along which the distribution
+        is evaluated.
+
+    Returns
+    -------
+    fn : array (N,)
+        Sampled Bingham function values at each point on directions.
+
+    Notes
+    -----
+    The Bingham function is defined as [1]_,[2]_,[3]_:
+
+    .. math::
+
+        f(n) = f_0 \exp[-k_1(\mu_1^Tn)^2-k_2(\mu_2^Tn)^2]
+
+    where $f(n)$ is the Bingham function value at a given direction $n$, $f_0$
+    is the Bingham maximum peak value, $k_1$ and $k_2$ are the concentration
+    constants parameters (large values correspond to lower dispersion values)
+    along the two dispersion axes $\mu_1$ and $\mu_2$.
+
+    References
+    ----------
+    .. [1] Bingham, C., 1974. An antipodally symmetric distribution on the
+           sphere. Anna1 Stat. 2, 1201-1225.
+    .. [2] Riffert, T.W., Schreiber, J., Anwander, A., Knösche, T.R., 2014.
+           Beyond fractional Anisotropy: Extraction of Bundle-specific
+           structural metrics from crossing fiber models.
+           Neuroimage 100: 176-191. doi: 10.1016/j.neuroimage.2014.06.015
+    .. [3] Henriques RN, 2018. Advanced Methods for Diffusion MRI Data Analysis
+           and their Application to the Healthy Ageing Brain (Doctoral thesis).
+           Downing College, University of Cambridge. doi: 10.17863/CAM.29356
+    """
+    if not (np.linalg.norm(vertices, axis=-1) == 1).any():
```

Comment:
Perhaps we should consider removing this warning here given that this function is being used at a low level, e.g. this if statement is being repeatedly evaluated for all voxels in functions such as "bingham_from_odf" and "bingham_from_sh".

Reply:
I did like we discussed, that is, I added a `_bingham_to_odf` internal method in which the check is not done.